contracts/interfaces/IAINFT.sol

```diff
@@ -0,0 +1,23 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+// import "@openzeppelin/contracts/interfaces/IERC721Enumerable.sol";
+
+//TODO: implement the function inside
+interface IAINFT {
+
+    function isApprovedOrOwner(address spender, uint256 tokenId) external view returns (bool);
```

Comment:
What about `isApprovedOrOwnerForUpdate`

Reply:
Is there any reason for distinguishing and renaming this function to `isApprovedOrOwnerForUpdate`? I just wonder.

Reply:
`_isApprovedOrOwner(address spender, uint256 tokenId) → bool` was already existed on ERC721.
https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721-_isApprovedOrOwner-address-uint256-
So I want to divide this to that.